### DISCUSSION TITLE
[Rust] Not implemented on Run

### DISCUSSION THREAD

**Discussion by red-avtovo** [coder/code-server#2065] [Q&A]
Hi! First of all, I would like to thank you for such a great project!
I'm experimenting with developing on Rust with code-server and I'm facing a weird(to me) issue.
When rust-analyzer extension successfully scans through the project it gives me an ability to run tests:
![Screenshot from 2020-09-05 22-13-13](https://user-images.githubusercontent.com/10682826/92312855-3a10d980-efc5-11ea-9372-e3fdf8738f09.png)
Normally it should run 'cargo test' but here I get the error:
![Screenshot from 2020-09-05 22-13-28](https://user-images.githubusercontent.com/10682826/92312868-53198a80-efc5-11ea-9541-fdd783da634e.png)
with the following stack trace:

```
ERR Not implemented: $getDefaultShellAndArgs@http://localhost:8080/static/de41646fc402b968ca6d555fdf2da7de9554d28a/usr/lib/code-server/lib/vscode/out/vs/workbench/services/extensions/worker/extensionHostWorker.js:768:607
_doInvokeHandler@http://localhost:8080/static/de41646fc402b968ca6d555fdf2da7de9554d28a/usr/lib/code-server/lib/vscode/out/vs/workbench/services/extensions/worker/extensionHostWorker.js:729:202
_invokeHandler@http://localhost:8080/static/de41646fc402b968ca6d555fdf2da7de9554d28a/usr/lib/code-server/lib/vscode/out/vs/workbench/services/extensions/worker/extensionHostWorker.js:728:901
_receiveRequest@http://localhost:8080/static/de41646fc402b968ca6d555fdf2da7de9554d28a/usr/lib/code-server/lib/vscode/out/vs/workbench/services/extensions/worker/extensionHostWorker.js:727:545
_receiveOneMessage@http://localhost:8080/static/de41646fc402b968ca6d555fdf2da7de9554d28a/usr/lib/code-server/lib/vscode/out/vs/workbench/services/extensions/worker/extensionHostWorker.js:726:373
m/<@http://localhost:8080/static/de41646fc402b968ca6d555fdf2da7de9554d28a/usr/lib/code-server/lib/vscode/out/vs/workbench/services/extensions/worker/extensionHostWorker.js:724:476
fire@http://localhost:8080/static/de41646fc402b968ca6d555fdf2da7de9554d28a/usr/lib/code-server/lib/vscode/out/vs/workbench/services/extensions/worker/extensionHostWorker.js:46:222
d/<@http://localhost:8080/static/de41646fc402b968ca6d555fdf2da7de9554d28a/usr/lib/code-server/lib/vscode/out/vs/workbench/services/extensions/worker/extensionHostWorker.js:857:513
```

The same happens if I trigger `cargo build` from build tasks in command pallet.
I'm not sure that it is a bug or just the absence of the feature I would happy to have but I would be glad if you will unlock that feature and let me build Rust projects not just from the terminal :smiley:

**Comment by bpmct** [comment]
cc @code-asher

**Comment by code-asher** [comment]
It appears to be a deliberate error but it looks like it might be pretty easy to patch. I'll try it out soon.

**Comment by nhooyr** [comment]
Perhaps this should be an issue.

**Comment by nhooyr** [comment]
Filed https://github.com/cdr/code-server/issues/2165

Thanks for reporting @red-avtovo
